src/renderer/components/CodeEditor/SqlCodeEditor.tsx

```diff
@@ -6,20 +6,94 @@ import {
   acceptCompletion,
   completionStatus,
   startCompletion,
+  autocompletion,
+  CompletionContext,
+  CompletionResult,
 } from '@codemirror/autocomplete';
 import { defaultKeymap, insertTab } from '@codemirror/commands';
 import { keymap } from '@codemirror/view';
-import { SQLConfig, sql, MySQL } from '@codemirror/lang-sql';
-import { Ref, forwardRef } from 'react';
+import { Ref, forwardRef, useCallback } from 'react';
 import useCodeEditorTheme from './useCodeEditorTheme';
+import type { EnumSchema } from 'renderer/screens/DatabaseScreen/QueryWindow';
+import { SyntaxNode } from '@lezer/common';
+import {
+  SQLConfig,
+  sql,
+  MySQL,
+  genericCompletion,
+  keywordCompletionSource,
+  schemaCompletionSource,
+} from 'query-master-lang-sql';
 
 const SqlCodeEditor = forwardRef(function SqlCodeEditor(
-  props: ReactCodeMirrorProps & { schema: SQLConfig['schema'] },
+  props: ReactCodeMirrorProps & {
+    schema: SQLConfig['schema'];
+    enumSchema: EnumSchema;
+  },
   ref: Ref<ReactCodeMirrorRef>
 ) {
-  const { schema, ...codeMirrorProps } = props;
+  const { schema, enumSchema, ...codeMirrorProps } = props;
   const theme = useCodeEditorTheme();
 
+  const enumCompletion = useCallback(
+    (context: CompletionContext, tree: SyntaxNode): CompletionResult | null => {
+      // dont run if there is no enumSchema
+      if (enumSchema.length === 0) return null;
+
+      const isEqualOperator = tree.prevSibling;
+
+      // check if it's an operator and id is 15
+      if (
+        !(
+          isEqualOperator?.type.name === 'Operator' &&
+          isEqualOperator?.type.id === 15
```

Comment:
Is the ID of the `=` always 15? I think it is better to do.

```
context.doc.sliceString(isEqualOperator.from, isEqualOperator.to) === '='
```

And also remember that '=' and '!=' also trigger the ENUM auto completion

Reply:
the != is also considered as two separate node 💆‍♂️

Reply:
I have changed the logic to explicitly check for '=' and '!='